encoding/thrift/register.go

```diff
@@ -79,6 +79,10 @@ type Service struct {
 	Name          string
 	Methods       map[string]UnaryHandler
 	OnewayMethods map[string]OnewayHandler
+
+	// Snippet of Go code representing the function definition of the different
+	// handlers. This is useful for introspection.
+	Signatures map[string]string
```

Comment:
To me, this is an odd place to stash this information. I'd expect each `thrift.UnaryHandler` and `thrift.OnewayHandler` to implement `Signature() string`.